### PR TITLE
Add token auth docs

### PR DESCRIPTION
**Pull request by BernhardPosselt**

### CODE REVIEW

File: docs/restapi.rst [hunk 1]
```diff
@@ -5,6 +5,19 @@ A REST API for publishing and deleting app releases has been built into the stor
...
+
+Several routes require authentication. The following authentication methods are supported:
+
+* **Basic**: Http header where **CREDENTIALS** is ``base64encode('user:password')``::
+
+    Authorization: Basic CREDENTIALS
+
+* **Basic**: Http header where **TOKEN** is a token which can be looked up in your profile::
+
+    Authorization: Bearer TOKEN
```
Comment (adsworth) [inline docs/restapi.rst]: It's Authorization: Token TOKEN, though that can be changed to Bearer. We would need to subclass TokenAuthentication and set the keyword class attribute. See also http://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication

File: docs/restapi.rst [hunk 2]
```diff
@@ -5,6 +5,19 @@ A REST API for publishing and deleting app releases has been built into the stor
...
+Authentication
+--------------
+
+Several routes require authentication. The following authentication methods are supported:
+
+* **Basic**: Http header where **CREDENTIALS** is ``base64encode('user:password')``::
+
+    Authorization: Basic CREDENTIALS
+
+* **Basic**: Http header where **TOKEN** is a token which can be looked up in your profile::
```
Comment (janibonnevier) [inline docs/restapi.rst]: I believe this line should say **Token** instead of **Basic**.

File: docs/restapi.rst (hunk 1 above)
Comment (BernhardPosselt) [inline docs/restapi.rst]: Remembered it wrongly, its token now :)

File: docs/restapi.rst (hunk 2 above)
Comment (BernhardPosselt) [inline docs/restapi.rst]: damn xD

**Pull request merged**